pkg/manager/repo_indexer/cron.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 The OpenPitrix Authors. All rights reserved.
+// Use of this source code is governed by a Apache license
+// that can be found in the LICENSE file.
+
+package repo_indexer
+
+import (
+	"context"
+
+	"github.com/robfig/cron"
+
+	repoClient "openpitrix.io/openpitrix/pkg/client/repo"
+	"openpitrix.io/openpitrix/pkg/config"
+	"openpitrix.io/openpitrix/pkg/constants"
+	"openpitrix.io/openpitrix/pkg/logger"
+	"openpitrix.io/openpitrix/pkg/pb"
+	"openpitrix.io/openpitrix/pkg/utils/sender"
+)
+
+type repoInfo struct {
+	repoId string
+	owner  string
+}
+
+type repoInfos []*repoInfo
+
+func getRepos() (*repoInfos, error) {
+	ctx := sender.NewContext(context.Background(), sender.GetSystemUser())
+	repoManagerClient, err := repoClient.NewRepoManagerClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	limit := uint32(50)
+	offset := uint32(0)
+	var rs repoInfos
+	for {
+		req := pb.DescribeReposRequest{
+			Limit:  limit,
+			Offset: offset,
+			Status: []string{constants.StatusActive}}
+		res, err := repoManagerClient.DescribeRepos(ctx, &req)
+		if err != nil {
+			return nil, err
+		}
+		for _, r := range res.GetRepoSet() {
+			rs = append(rs, &repoInfo{
+				repoId: r.GetRepoId().GetValue(),
+				owner:  r.GetOwner().GetValue(),
+			})
+		}
+		if len(res.GetRepoSet()) >= int(limit) {
```

Comment:
If length is bigger than 50, the item has already been appended into rs.

Reply:
Yes, this func will return all of the repo info.